raster-foundry/raster-foundry
# Fix export status update

## Overview

This PR fixes how export statuses get updated in `rf update`. There were a few things that
hurt a bit that get fixed here. More on that in notes. Also it makes it so you can't stitch exports
anymore, because we're doing it in post.

### Checklist

- [x] ~Any new SQL strings have tests~ Untested sql strings that didn't work were corrected (again, see notes)

### Notes

Oh boy was there some fun stuff in here.

#### Part 1 -- why aren't failure statuses being written correctly?

They were! We were just writing the [wrong status](https://github.com/raster-foundry/raster-foundry/compare/feature/js/feature/js/exit-correctly-on-export-failure?expand=1#diff-6ddd036f93253f31a91daa93c5920506L349). But everything basically worked.

#### Part 2 -- why doesn't `ExportDao.update` work?

an `owner = ...` filter. This was the only place in the codebase where the update method hadn't been
updated to reflect the fact that we do authorization separately, and because it was at the end of a [big
sql string](https://github.com/raster-foundry/raster-foundry/compare/feature/js/feature/js/exit-correctly-on-export-failure?expand=1#diff-c6e6ab62b2fdb4048a49e18d0a6f17b1L67),
it took me a while to notice. Probably [the line in `CheckExportStatus`](https://github.com/raster-foundry/raster-foundry/compare/feature/js/feature/js/exit-correctly-on-export-failure?expand=1#diff-2be41e16df351239487d4d36ff569765L150)
(removed) that rewrote the sql to update the status should have been a tip off that something weird
was afoot.

#### Part 3 -- why won't my logging messages from running the export print?

I have _no idea_. I left that unresolved. I'd like the debug info to be available if we want it, but the logs
just won't print.

## Testing Instructions

 * in the `for` comprehension to do exports, before anything time consuming happens, throw in a new
   `_ <- IO(throw new Exception("for the lulz"))`
 * reassemble your batch jar
 * rebuild your batch container
 * create an export of a small area of a scene without many bands (the Vermont image is good for this)
 * `./scripts/console batch 'rf export <the id for that export>`
 * it should fail, tell you that it fails, and set the export status to FAILED for that export
 * remove your exception
 * reassemble your batch jar
 * run the export again
 * it should succeed, tell you that it succeeded, and set the export status to EXPORTED for that export
 * download the file and check it out

Closes #3970 

Closes #3971
